atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -149,6 +156,55 @@ def get_buildroot(self, worker_metadatas):
 
         return buildroots
 
+    def get_logs(self):
+        """
+        Build list of log files
+
+        :return: list, of log files
+        """
+
+        logs = None
+        output = []
+
+        # Collect logs from server
+        try:
+            logs = self.osbs.get_orchestrator_build_logs(self.build_id)
+        except OsbsException as ex:
+            self.log.error("unable to get build logs: %r", ex)
+            return output
+        except TypeError:
+            # Older osbs-client has no get_orchestrator_build_logs
+            self.log.error("Your osbs client doesn't support this functionality")
+            return output
+
+        platform_logs = {}
+        for entry in logs:
+            platform_logs.setdefault(entry.platform, [])
+            platform_logs[entry.platform].append(entry.line)
```

Comment:
Should we be concerned with loading the whole log into memory?
Not sure if writing each line per iteration is needed.

Reply:
A prior iteration of the code did loop through the log just trying to build a list of arches, and then wrote out the log lines to the files, but it ended up looping through the logs once to build the list of arches, then one time for each of the arches as well, and @twaugh actually suggested the above. I suppose one question that should be asked here is if we should be trying to dynamically build the list of arches, just assume a static list of possible arches, and not send any empty logs, so we only need to loop over the combined log once to split out each line and write them out to individual files. Might be possible to do all in a single loop, including creating a dynamic arch list, by someone a bit more adept at python than I am... :)

Reply:
I think it can be done in a single loop. How about `.setdefault(entry.platform, NamedTemporaryFile(...))`...?

Reply:
Okay, per discussion on IRC, I think I've whipped up something that seems to work with a single pass through the logs, and without loading the whole log into memory before writing it out to temp files.